frontend/public/components/factory/table.tsx

```diff
@@ -304,7 +307,7 @@ export const Table = connect<TablePropsFromState,TablePropsFromDispatch,TablePro
       staticFilters: PropTypes.array,
       virtualize: PropTypes.bool,
       currentSortField: PropTypes.string,
-      currentSortFunc: PropTypes.string,
+      currentSortFunc: PropTypes.string || PropTypes.func,
```

Comment:
You're overloading `sortFunc` to mean completely different things depending on the type that's passed. As mentioned in the previous PR, this should be a separate prop specfically to transform the value for sorting.

Reply:
idk, I feel it's appropriate.  When it's a string, it's an index into an array of sortFuncs, when it's a func, it's a direct sortFunc as defined by the app dev.   If I add another property, it seems more confusing to add a `sortConversion` which is the same as `sortFunc`.

Reply:
>  If I add another property, it seems more confusing to add a sortConversion which is the same as sortFunc.

But it's not the same as `sortFunc`! They're fundamentally different. `sortFunc` gets the value out of the object. `sortConversion` transforms the `sortField` value.

Today you never pass both `sortField` and `sortFunc` together. You've created a special case where if you pass *both* `sortField` and `sortFunc` AND `sortFunc` is a function, it does something special. That's a confusing and unintuitive API. The only way you would know this is to read the code closely (and the code is already a little hard to understand).

Honestly, I feel like we're overthinking this. Why not just pass an optional `sortAsNumber` prop to tell table to treat the column as a number. That's what we need here. If we ever need something more general, we can revisit.

Reply:
ok, it doesn't seem all that confusing to me :-)  if you specify sortField, that field will be sorted as a string, if you pass sortFunc:string, you are defining a named sortFunction (meaning the function knows what field to sort on, it's in the func name,ex: "getClusterOperatorStatus"), if you define a sortField and sortFunc:func, your sortFunc will be called passing the sortField.

`sortAsNumber` seems like a good solution, I'll implement that -thanks

Reply:
implemented 'sortAsNumber'.